src/util/alloc/large_object_allocator.rs

```diff
@@ -36,21 +36,20 @@ impl<VM: VMBinding> Allocator<VM> for LargeObjectAllocator<VM> {
 
     fn alloc(&mut self, size: usize, align: usize, offset: isize) -> Address {
         #[cfg(debug_assertions)]
-        crate::util::alloc::object_ref_guard::assert_object_ref_in_cell::<VM>(size);
+        crate::util::alloc::object_ref_guard::assert_object_ref_in_allocation::<VM>(size);
```

Comment:
To my understanding, this assertion should always be true and is redundant. The only case for JikesRVM to have an object pointer pointing outside of the object is zero-sized objects. For these zero-sized objects, (1) they should not go to LOS (2) even if they do, they will never be placed at the end of a chunk or page, since LOS allocate memory at page granularity.

Reply:
This is removed in https://github.com/mmtk/mmtk-core/pull/699.